clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureComputeClient.groovy

```diff
@@ -65,6 +68,22 @@ public class AzureComputeClient extends AzureBaseClient {
     ComputeManagementClient computeClient = new ComputeManagementClientImpl(tokenCredentials)
     computeClient.setSubscriptionId(this.subscriptionId)
     computeClient.setLogLevel(HttpLoggingInterceptor.Level.NONE)
+
+    // Add Azure Spinnaker telemetry capturing
+    computeClient.getClientInterceptors().add(new Interceptor() {
+      @Override
+      Response intercept(Interceptor.Chain chain) throws IOException {
+        Request.Builder builder = chain.request().newBuilder()
+        def oldHeaderValue = chain.request().header("User-Agent")
+        if (oldHeaderValue) {
+          builder.header("User-Agent", "AzureSpinnaker ${oldHeaderValue}")
+        } else {
+          builder.header("User-Agent", "AzureSpinnaker")
+        }
```

Comment:
I would simplify like this:
userAgentValue = oldHeaderValue ? "AzureSpinaker" : "AzureSpinnaker ${oldHeaderValue}";
builder.header("User-Agent", userAgentValue);